### ISSUE TITLE
No auto completion for modules

### ISSUE THREAD

**Issue by GideonUng** [swift-server/vscode-swift#325]
**Describe the bug**
no auto completion between modules

on hover of a function sourcekit lsp logs:
```
2022-06-06 08:30:42.511 sourcekit-lsp.exe[11968:4768] {
  key.internal_diagnostic: "Resolved to incomplete expression or statement."
}
2022-06-06 08:30:42.511 sourcekit-lsp.exe[11968:4768] SourceKitServer: Response<textDocument/hover(20)>(
  success(nil)
)
```

**To Reproduce**
create a package with 2 modules where one depends on the other

**Expected behavior**
auto completion

**Environment**
 - OS: Ubuntu 22.04, Windows 11
 - Swift version 5.6.1
 - Visual Studio Code version: latest
 - vscode-swift version: latest

**Additional context**

**Comment by adam-fowler** [comment]
Did you compile everything before running this test? Sourcekit-LSP needs you to compile everything at least once before it catches symbols from other modules

**Comment by GideonUng** [comment]
yes

**Comment by GideonUng** [comment]
on ubunut 22.04 it seems to work now. still not for win 11

**Comment by adam-fowler** [comment]
> on ubunut 22.04 it seems to work but only with swift 5.7. still not for win 11

Was just about to ask if you tried the 5.7 or later. There are fixes in the 5.7 version of sourcekit-lsp which catch changes to the Package.swift manifest while you are running. With 5.6 it should work if you restart VSCode (or running command `Developer: Reload Window`) whenever changing Package.swift.

Also to note swift doesn't have a official 22.04 release of 5.6, you can only get 22.04 targeted releases as trunk snapshots at the moment.

**Comment by GideonUng** [comment]
5.7 also does not work on windows with the exact same issue. Reloading the window does nothing. Neither does building the project.

**Comment by adam-fowler** [comment]
It was working, I'll have to check and see if something in 5.7 windows has broken it again. 
If you go to the output pane, in the drop down is there a Swift entry?. If so can you add the contents of the window here.
Can you run `swift build` outside of VSCode with no issue?

**Comment by GideonUng** [comment]
yes building is no issue at all. debugging does not work on windows or linux but thats another issue.

**Comment by adam-fowler** [comment]
Do you have those logs from the swift output pane? One for Linux and one for Windows. It'll give me a better idea where you are seeing failures.

**Comment by GideonUng** [comment]
what logs? the one from the sourcekit lsp?

**Comment by adam-fowler** [comment]
Sorry, open VSCode, load a Package.swift, open a swift file and go to the output view, select Swift from the drop down.

**Comment by GideonUng** [comment]
yeah nothing in there
```
11:52:28: compnerd.org Swift version 5.7-dev (LLVM 9fde71b269b42b8, Swift ddc46e1f5bd885d)
11:52:29: SourceKit-LSP setup for ST
11:52:29: ST: add: c:\Users\gidch\Desktop\ST
11:52:29: ST: Resolving Dependencies ... done.
11:52:31: ST: focus: c:\Users\gidch\Desktop\ST
```

**Comment by GideonUng** [comment]
Linux does seem to work. I think my issue on Linux was only related to swift root not being found through path and me setting swift root manually has "fixed" it for now.
Enjoy your vacation!

**Comment by adam-fowler** [comment]
> Linux does seem to work. I think my issue on Linux was only related to swift root not being found through path and me setting swift root manually has "fixed" it for now.
> Enjoy your vacation!

It should display an error when it cannot find the swift binaries. I think I need to make that clearer

**Comment by GideonUng** [comment]
They are in path so I'm not sure why they aren't being found. And i didn't see a notification for it. I think it was just a silent fail which is why I attributed it to the same issue as on windows.

**Comment by adam-fowler** [comment]
Would you be able to paste the output from the swift output pane on Linux when you don't set the path to the swift executable.

**Comment by adam-fowler** [comment]
Lets just leave it for the moment. If you see it again later, I guess you can report it again

**Comment by GideonUng** [comment]
![image](https://user-images.githubusercontent.com/5054701/191284883-eec4bcec-8957-4767-a85e-e309df415d8a.png)
still persistent with newest versions and vanilla setup. compiles just fine

**Comment by adam-fowler** [comment]
Is this Ubuntu 22.04 on windows?

**Comment by GideonUng** [comment]
windows
